docs/new-architecture-app-modules-android.md

```diff
@@ -0,0 +1,415 @@
+---
+id: new-architecture-app-modules-android
+title: Enabling TurboMudule on Android
+---
+
+Make sure your application meets all the [prerequisites](new-architecture-app-intro).
+
+## 1. Enable NDK and the native build
+
+:::caution
+
+In this iteration of the playbook we’re setting up the project to let you build from source. You might notice an increase in your build time because of this. We’re looking into what would be the preferred approach here so please feel free to share your feedbacks.
+
+:::
+
+The code-gen will output some Java and some C++ code that now we need to build.
+
+Let’s edit your **module level** `build.gradle` to include the **two** `externalNativeBuild` blocks detailed below inside the `android{}` block:
+
+```groovy
+android {
+    defaultConfig {
+        applicationId "com.awesomeproject"
+        // ...
+
+        // Add this block
+        externalNativeBuild {
+            ndkBuild {
+                arguments "APP_PLATFORM=android-21",
+                        "APP_STL=c++_shared",
+                        "NDK_TOOLCHAIN_VERSION=clang",
+                        "GENERATED_SRC_DIR=$buildDir/generated/source",
+                        "PROJECT_BUILD_DIR=$buildDir",
+                        "REACT_ANDROID_DIR=$rootDir/../node_modules/react-native/ReactAndroid",
```

Comment:
I am using react-native@0.66.4 and For supporting new architecture in my app, I have to set these two variables in `build.gradle` like this

![image](https://user-images.githubusercontent.com/36044436/149882345-11ea65a6-46ae-4f2b-b2aa-ab7bf2b3af1b.png)

```
ANDROID_NDK_VERSION = "21.4.7075529"
ANDROID_NDK_PATH="/Users/apple/Library/Android/sdk/ndk/21.4.7075529"
```

and getting this error

fcntl(): Bad file descriptor
/Users/apple/Library/Android/sdk/ndk/21.4.7075529/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting    .  Stop.

Besides this, I also have to copy `react-native/packages/react-native-codegen/android` folder to `../node_modules/react-native-codegen/android`
to make it run properly

Reply:
> I am using react-native@0.66.4

The whole playbook relies on using RN nightlies. RN 0.66.x won't work as some of the infrastructure code is missing.